Observe MassFlux in EvolveValenciaDivClean

markscheel opened the pull request:
## Proposed changes

Adds the interpolator to EvolveValenciaDivClean.

FishboneMoncriefDisk now observes MassFlux.

The other EvolveValenciaDivClean executables do not currently have interpolators,
but there is an easy mechanism for adding them.

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

markscheel committed:
EvolveValenciaDivClean: Add interpolators.

Now FishboneMoncriefDisk observes MassFlux.

markscheel commented:
Tests failed before because I hadn't rebased onto develop. I did so, and fixed what didn't compile.

markscheel commented:
clang-format change only.

markscheel commented:
Rebased.

I separated 3 of the commits here into separate PRs that must be merged first, before this one.

markscheel committed:
Allow Variables<tmpl::list<>>.

Useful for specifying that no variables will be interpolated
(i.e. the interpolator is not used) in an executable.

nilsdeppe commented on src/Evolution/Executables/GrMhd/ValenciaDivClean/CMakeLists.txt:
```diff
@@ -2,6 +2,7 @@
 # See LICENSE.txt for details.
 
 set(LIBS_TO_LINK
+  ApparentHorizons
```
These aren't needed for all the executables

markscheel replied:
So I should just remove them?  Does cmake link them through another mechanism?

kidder commented on src/Evolution/Executables/GrMhd/ValenciaDivClean/EvolveValenciaDivClean.hpp:
```diff
@@ -98,7 +115,31 @@ class CProxy_ConstGlobalCache;
+      hydro::Tags::LorentzFactor<DataVector>, gr::Tags::Lapse<DataVector>,
+      gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+      gr::Tags::SqrtDetSpatialMetric<DataVector>>;
+  using compute_items_on_target = tmpl::append<
```
tmpl::push_front<list,A,B>

markscheel committed:
Add OptionTags::EventsAndTriggers::name().

This fixes a hard-to-find error that arises for
sufficiently complicated template arguments to
OptionTags::EventsAndTriggers.

kidder commented:
I am happy for you to squash if @nilsdeppe is

nilsdeppe reviewed (commented):
I'm happy for you to squash

markscheel committed:
AddSpectreExecutable: target_link_libraries PUBLIC.

This allows other CMakeLists.txt to add extra libraries
to target_link_libraries.

markscheel commented:
Squashed, and added another commit changing AddSpectreExecutable.cmake

nilsdeppe reviewed (approved):
![image](https://user-images.githubusercontent.com/13531030/76245892-1980e600-6213-11ea-88cd-1b282afc1d04.png)
